### PR TITLE
filter events if the trace is not sampled

### PR DESCRIPTION
**Pull request by Geal**
there's no need to record the events in opentelemetry if the trace will not be registered

**Checklist**

Complete the checklist (and note appropriate exceptions) before a final PR is raised.

- [ ] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]. It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]. Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]. Tick whichever testing boxes are applicable. If you are adding Manual Tests:
    - please document the manual testing (extensively) in the Exceptions.
    - please raise a separate issue to automate the test and label it (or ask for it to be labeled) as `manual test`

### CODE REVIEW

File: apollo-router/src/plugins/telemetry/reload.rs [hunk 1]
```diff
@@ -166,20 +166,18 @@ where
...
-                .map(|id| cx.span(id).is_some())
-                // - there's no parent span (it's the root), so we make the sampling decision
-                .unwrap_or_else(|| self.sample())
+        // this span is enabled if:
+        current_span
+            .id()
+            // - there's a parent span and it was enabled
+            .map(|id| cx.span(id).is_some())
+            // - there's no parent span (it's the root), so we make the sampling decision
+            .unwrap_or_else(|| self.sample())
```
Comment (garypen) [inline apollo-router/src/plugins/telemetry/reload.rs:180]: I don't really understand this filtering, but it feels like it should be:
```
meta.is_span() &&
current_span
            .id()
            // - there's a parent span and it was enabled
            .map(|id| cx.span(id).is_some())
            // - there's no parent span (it's the root), so we make the sampling decision
            .unwrap_or_else(|| self.sample())
```
I don't understand why we wouldn't include the metadata in the `enabled` decision since it would be a quick exit if it isn't a span.

File: apollo-router/src/plugins/telemetry/mod.rs [hunk 2]
```diff
@@ -613,18 +615,34 @@ impl Telemetry {
...
+        if tracing_config.jaeger.is_none()
+            && tracing_config.zipkin.is_none()
+            && tracing_config.datadog.is_none()
+            && tracing_config.otlp.is_none()
+            && config.apollo.is_none()
+        {
+            sampler = SamplerOption::Always(Sampler::AlwaysOff);
+        }
+        // FIXME: add a test to set to 0 if none are configured
+        SamplingFilter::configure(&sampler);
```
Comment (BrynCooke) [inline apollo-router/src/plugins/telemetry/mod.rs:644]: I think this needs to be done in activate?

**Comment by BrynCooke** [comment]
Let's make sure we have tests which use a 0.5 sample rate to ensure that we never get half traces.

File: apollo-router/src/plugins/telemetry/mod.rs (hunk 2 above)
Comment (Geal) [reply apollo-router/src/plugins/telemetry/mod.rs:652]: activate is when we actually start using the plugin, right?

**Review by bnjjj — APPROVED** [review]

**Comment by Geal** [comment]
> Let's make sure we have tests which use a 0.5 sample rate to ensure that we never get half traces.
[84c6558](https://github.com/apollographql/router/pull/2999/commits/84c6558f9e2a8f0cb00b240f6e61753fcdaa4f4e)

File: apollo-router/src/plugins/telemetry/mod.rs (hunk 2 above)
Comment (Geal) [reply apollo-router/src/plugins/telemetry/mod.rs]: done in [b3cd920](https://github.com/apollographql/router/pull/2999/commits/b3cd920b9b07626a872360cce538c72f8e61f3a0)

File: apollo-router/src/plugins/telemetry/reload.rs (hunk 1 above)
Comment (Geal) [reply apollo-router/src/plugins/telemetry/reload.rs]: it's been done, thanks for the suggestion

**Review by BrynCooke — APPROVED** [review]
